File: exercism/go/isbn-verifier/isbn_verifier.go
package isbn

import (
	"unicode"
)

func IsValidISBN(isbn string) bool {
	total := 0
	multi := 10

	for _, value := range isbn {
		if unicode.IsNumber(value) {
			// get rune value out of the ASCII
			total += int(value-'0') * multi
			multi--
		} else if value == '-' {
			continue
		} else if value == 'X' && multi == 1 {
			total += +10 * multi
			multi--
		} else {
			return false
		}
	}

	return total%11 == 0 && multi == 0
}

File: exercism/python/isbn-verifier/isbn_verifier.py
"""
ISBN Verifier
"""


def is_valid(isbn):
    """
    ISBN-10 verification process

    using a replace a re.match and
        digits = [value == 'X' and 10 or int(value) for value in isbn]
        summ = sum(value * (10 -  index) for index, value in enumerate(digits))
    doesn't seem as efficient
    """
    total = 0
    multi = 10

    for value in isbn:
        if value.isdigit():
            total = total + int(value) * multi
            multi -= 1
        elif value == '-':
            continue
        elif value == 'X' and multi == 1:
            total = total + 10 * multi
            multi -= 1
        else:
            return False

    return total % 11 == 0 and not multi
